src/resolvers/products/index.ts

```diff
@@ -33,5 +37,39 @@ export const ProductResolver = {
 
       return product;
     },
+    updateProduct: async (_: never, data: UpdateProductInput, ctx: Context) => {
+      const user: Account = ctx.data;
+      const { input } = data;
+      const { id, body } = input;
+      const { name, description } = body;
+
+      if (name === '') throw new UserInputError('Please enter product name.');
+      if (description === '')
+        throw new UserInputError('Please enter product description.');
+
+      const product = await ProductModel.findOne({
+        id: id,
+        owner: user.id,
+      });
+
+      if (!product) throw new UserInputError('Product does not exist');
+
+      const toUpdate: LooseObject = {
+        ...body,
+      };
+
+      if (name && name !== '') {
```

Comment:
In here, you can already remove checking of `name !== ''` since you already throw an error above when `name === ''`.